monkey/infection_monkey/island_api_client/i_island_api_client.py

```diff
@@ -60,7 +60,7 @@ def get_pba_file(self, filename: str) -> bytes:
         """
 
     @abstractmethod
-    def get_agent_binary(self, operating_system: OperatingSystem) -> Optional[bytes]:
+    def get_agent_binary(self, operating_system: OperatingSystem) -> bytes:
```

Comment:
Should we be using `Optional` here? The only time it won't return `bytes` is when an error is raised.

Reply:
See [this SO question on type hinting with exceptions]( https://stackoverflow.com/a/44282299). Maybe we should be using [typing.NoReturn](https://docs.python.org/3/library/typing.html#typing.NoReturn).

Reply:
`Optional` means the value could be `None`. That is not the case here. I don't think we need to do anything special here because it is expected that we won't get a return value if the method raises an exception. As long as we've documented the exceptions that are raised, I think we're covered here

Reply:
I just realised I wrote that comment thinking you added `Optional` to the type hint. 😅 
I agree with you. Documenting the exception feels enough.